src/main/java/com/kodilla/ecommercee/service/CartDbService.java

```diff
@@ -0,0 +1,108 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.*;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.repository.CartRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CartDbService {
+
+    private CartRepository cartRepository;
+    private CartMapper cartMapper;
+    private OrderDbService orderDbService;
+    private ProductMapper productMapper;
+
+    @Autowired
+    public CartDbService(CartRepository cartRepository, CartMapper cartMapper,
+                         OrderDbService orderDbService, ProductMapper productMapper) {
+        this.cartRepository = cartRepository;
+        this.cartMapper = cartMapper;
+        this.orderDbService = orderDbService;
+        this.productMapper = productMapper;
+    }
+
+    public void saveCart(Cart cart) {
+        cartRepository.save(cart);
+    }
+
+    public Optional<Cart> getCartById(Long cartId) {
+        return cartRepository.findById(cartId);
+    }
+
+    public List<Cart> getCarts() {
+        return cartRepository.findAll();
+    }
+
+    public CartDto createEmptyCart(Long userId) {
+        List<ProductDto> items = new ArrayList<>();
+        return CartDto.builder().id(null).totalPrice(new BigDecimal(0.0)).isClosed(false)
+                .userId(userId).cartItems(items).build();
+    }
+
+    public List<ProductDto> getProductsFormBasket(Long cartId) {
```

Comment:
```suggestion
    public List<ProductDto> getProductsFromCart(Long cartId) {
```
Dlaczego Basket, a nie Cart? Wydaje mi się, że powinniśmy trzymać się jednej nazwy dla koszyka, żeby kod był czytelny.

Reply:
Dobrze już ujednolicone.